edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerModule.cs

```diff
@@ -84,32 +84,31 @@ public virtual bool Equals(IModule<DockerConfig> other)
                 return true;
             }
 
-            return string.Equals(this.Name, other.Name) &&
+            return this.BaseEquals(other) &&
                 string.Equals(this.Version, other.Version) &&
-                string.Equals(this.Type, other.Type) &&
-                this.DesiredStatus == other.DesiredStatus &&
-                this.Config.Equals(other.Config) &&
-                this.RestartPolicy == other.RestartPolicy &&
-                this.ImagePullPolicy == other.ImagePullPolicy &&
-                this.StartupOrder == other.StartupOrder &&
-                this.IsEnvDictionaryEqual(other);
+                this.Config.Equals(other.Config);
         }
 
         public virtual bool IsOnlyModuleStatusChanged(IModule other)
         {
             return other is DockerModule dockerModule &&
-                string.Equals(this.Name, other.Name) &&
+                this.BaseEquals(other) &&
                 string.Equals(this.Version, other.Version) &&
+                this.Config.Equals(dockerModule.Config);
+        }
+
+        protected bool BaseEquals(IModule other)
```

Comment:
This seems a bit odd. Why do we need this method?

Reply:
@varunpuranik this can be reused in EdgeHubDockerModule.